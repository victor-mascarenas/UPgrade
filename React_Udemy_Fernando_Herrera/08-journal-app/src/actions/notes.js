import { type } from "@testing-library/user-event/dist/type";
import { db } from "../firebase/firebase-config";
import { types } from "../reducers/types/types";

export const startNewNote = () => {
    return async (dispatch, getState) => {
        const uid = getState().auth.uid;
        const newNote = {
            title: '',
            body: '',
            date: new Date().getTime()
        };
        const docRef = await db.collection(`${uid}/journal/notes`).add(newNote);
        dispatch(activeNote(docRef.id, newNote));
    };
};

export const activeNote = (id, note) => ({
    type: types.NOTES_ACTIVE,
    payload: {
        id,
        ...note
    }
});