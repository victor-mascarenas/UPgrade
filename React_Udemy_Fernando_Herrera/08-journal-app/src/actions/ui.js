import { types } from "../reducers/types/types";

export const setError = (msg) => {
    return ({
        type: types.UI_SET_ERROR,
        payload: {
            errorMsg: msg
        }
    });
};

export const removeError = () => {
    return ({
        type: types.UI_REMOVE_ERROR
    });
};