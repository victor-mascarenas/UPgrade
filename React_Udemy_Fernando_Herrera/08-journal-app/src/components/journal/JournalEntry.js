import React from 'react'

export const JournalEntry = () => {
    return (
        <div className='journal__entry'>
            <div className='journal__entry-picture'></div>
            <div className='journal__entry-body'>
                <p className='journal__entry-title'>
                    Un nuevo dia
                </p>
                <p className='journal__entry-content'>
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Id quod soluta nihil, in commodi.
                </p>
            </div>
            <div className='journal__entry-date-box'>
                <span>Monday</span>
                <h4>28</h4>
            </div>
        </div>
    )
}
