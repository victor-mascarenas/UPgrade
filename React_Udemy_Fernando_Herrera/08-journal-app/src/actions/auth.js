import { types } from "../reducers/types/types";
import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { endLoading, startLoading } from "./ui";
import Swal from 'sweetalert2';
import { notesLogout } from "./notes";

export const startLogin = (email, password) => {
    return (dispatch) => {//dispatch comes from Thunk
        dispatch(startLoading());
        return firebase.auth().signInWithEmailAndPassword(email, password)
            .then(({user}) => {
                dispatch(login(user.uid, user.displayName));
            })
            .catch(error => {
                Swal.fire('Error', error.message, 'error');
            })
            .finally(() => {
                dispatch(endLoading());
            });
    };
};

export const startCredentialsRegistration = (email, password, name) => {
    return (dispatch) => {
        firebase.auth().createUserWithEmailAndPassword(email, password)
            .then(async ({user}) => {
                await user.updateProfile({
                    displayName: name
                });
                dispatch(login(user.uid, user.displayName));
            })
            .catch(error => {
                Swal.fire('Error', error.message, 'error');
            });
    };
};

export const startGoogleLogin = () => {
    return (dispatch) => {
        firebase.auth().signInWithPopup(googleAuthProvider)
            .then(({user}) => {
                dispatch(login(user.uid, user.displayName));
            })
            .catch((error) => {
                Swal.fire('Error', error.message, 'error');
            });
    };
};

export const login = (uid, displayName) => (
    {
        type: types.LOGIN,
        payload: {
            uid,
            displayName
        }
    }
);

export const startLogout = () => {
    return async (dispatch) => {
        await firebase.auth().signOut();
        dispatch(logout());
        dispatch(notesLogout());
    };
};

export const logout = () => {
    return {
        type: types.LOGOUT
    };
}