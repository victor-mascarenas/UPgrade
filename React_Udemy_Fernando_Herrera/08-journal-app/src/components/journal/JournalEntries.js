import React from 'react'
import { JournalEntry } from './JournalEntry';

export const JournalEntries = () => {
    const entries = [1, 2, 3, 4];

    return (
        <div className='jurnal__entries'>
            {
                entries.map(value => (
                    <JournalEntry key={value}/>
                ))
            }
        </div>
    )
}
