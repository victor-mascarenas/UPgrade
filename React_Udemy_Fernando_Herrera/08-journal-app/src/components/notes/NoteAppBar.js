import React from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../actions/notes';

export const NoteAppBar = () => {
    const dispatch = useDispatch();
    const {active: note} = useSelector(state => state.notes);

    const fileInputStyle = {
        display: 'none'
    };

    const handleSave = () => {
        dispatch(startSaveNote(note));
    };
    const handlePicture = () => {
        document.querySelector('#fileSelector').click();
    };
    const handleFileChanged = (e) => {
        const file = e.target.files[0];
        if (file) {
            dispatch(startUploading(file));
        }
    };

    return (
        <div className='notes__appbar'>
            <span>28 de Agosto de 2022</span>
            <input id='fileSelector' name='file' type='file' style={fileInputStyle} onChange={handleFileChanged}/>
            <div>
                <button className='btn' onClick={handlePicture}>Add picture</button>
                <button className='btn' onClick={handleSave}>Save</button>
            </div>
        </div>
    )
}
