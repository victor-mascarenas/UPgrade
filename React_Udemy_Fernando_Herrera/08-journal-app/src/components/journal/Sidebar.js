import React from 'react'
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { startLogout } from '../../actions/auth';
import { JournalEntries } from './JournalEntries'

export const Sidebar = () => {
    const dispatch = useDispatch();
    const {name} = useSelector(state => state.auth);

    const handleLogOut = () => {
        dispatch(startLogout());
    };

    return (
        <aside className='journal__sidebar'>
            <div className='journal_sidebar-navbar'>
                <h3 className='mt-5'>
                    <i className='far fa-moon'/>
                    <span> {name}</span>
                </h3>
                <button className='btn' onClick={handleLogOut}>Logout</button>
            </div>
            <div className='journal__new-entry'>
                <i className='far fa-calendar-plus fa-5x'/>
                <p className='mt-5'>New entry</p>
            </div>
            <JournalEntries/>
        </aside>
    )
}
