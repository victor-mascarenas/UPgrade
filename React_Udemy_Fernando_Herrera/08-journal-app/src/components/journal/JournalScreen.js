import React from 'react'
import { useSelector } from 'react-redux'
import { NoteSection } from '../notes/NoteSection'
import { NothingSelected } from './NothingSelected'
import { Sidebar } from './Sidebar'

export const JournalScreen = () => {
    const {active} = useSelector(state => state.notes);

    return (
        <div className='journal__main-content animate__animated animate__fadeIn'>
            <Sidebar/>
            <main>
                {
                    (active) ?
                        <NoteSection/> :
                        <NothingSelected/>
                }
            </main>
        </div>
    )
}
