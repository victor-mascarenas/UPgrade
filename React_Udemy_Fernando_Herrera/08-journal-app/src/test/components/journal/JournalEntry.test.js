import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {};
const store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
    id: '10',
    title: 'hola',
    body: 'mundo',
    date: 0,
    url: 'https://algunlugar.com/foto.jpg'
}

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <JournalEntry note={note}/>
        </MemoryRouter>
    </Provider>
);

describe('Pruebas en JournalEntry', () => {
    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe disparar el activeNote', () => {
        wrapper.find('.journal__entry').simulate('click');
        expect(store.dispatch).toHaveBeenCalledWith(
            activeNote(note.id, {...note})
        );
    });
});