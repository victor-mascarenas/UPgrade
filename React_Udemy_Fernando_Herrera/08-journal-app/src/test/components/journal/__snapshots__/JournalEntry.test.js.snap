// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en JournalEntry Debe de mostrarse correctamente 1`] = `
<div
  className="journal__entry animate__animated animate__fadeIn"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://algunlugar.com/foto.jpg)",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      hola
    </p>
    <p
      className="journal__entry-content"
    >
      mundo
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31
    </h4>
  </div>
</div>
`;
